x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -183,21 +195,24 @@ public Authentication getAuthentication() {
     /**
      * Stores the initial response with the original headers of the authenticated user
      * and the expected expiration time.
+     * Currently for EQL we don't set limit for a stored async response
+     * TODO: add limit for stored async response in EQL, and instead of this method use createResponse
      */
-    public void createResponse(String docId,
+    public void createResponseForEQL(String docId,
```

Comment:
I wonder if we should introduce a new parameter that enables/disables the limit instead?

Reply:
@dnhatn  Are you suggesting the parameter for EQL only?  Some system variable?
I am a little bit concerned of introducing too many params, and the plan to enable this setting for EQL as well in the follow up where we ask EQL team for a review. WDYT?

Reply:
Confirm with @dnhatn offline, this is something we want to tackle in the follow up PRs.